back/lib/citizen_lab/lock_manager.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module CitizenLab
+  # Implements a (binary) lock on the top of PostgreSQL advisory locks.
+  # Inspired from: https://www.kostolansky.sk/posts/postgresql-advisory-locks/
+  module LockManager
```

Comment:
Probably just want to discuss what would happen if we got errors running code that has created a block. Does that lock get released automatically or do we end up with everything piling up behind it?

Reply:
The lock is also released if the transaction is rolled back.